DiscordBot.MessageHandlers/MessageResponsesHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Discord;
+using Discord.WebSocket;
+using DiscordBot.Core.Interfaces;
+using DiscordBot.MessageHandlers.Helpers;
+using Microsoft.Extensions.Logging;
+using System.Text.RegularExpressions;
+
+namespace DiscordBot.MessageHandlers
+{
+    public class MessageResponsesHandler : IMessageReceivedHandler
+    {
+        private readonly IDiscordClient _client;
+        private readonly ILogger<MessageResponsesHandler> _logger;
+        private readonly IRegexResponseHelper _regexResponseHelper;
+
+        public MessageResponsesHandler(ILogger<MessageResponsesHandler> logger, IDiscordClient client, IRegexResponseHelper regexResponseHelper)
+        {
+            _logger = logger;
+            _client = client;
+            _regexResponseHelper = regexResponseHelper;
+        }
+
+        private bool IsApplicable(SocketMessage socketMessage)
+            => socketMessage.Channel is SocketTextChannel &&
+            (socketMessage.MentionedUsers.Any(user => _client.CurrentUser.Mention.Equals(user.Mention)) ||
+            socketMessage.MentionedEveryone);
```

Comment:
@Niewidzialny84 This should also **not** be applicable if the author is a bot (`IUser.IsBot`)

Reply:
Resolving - the `MessageReceivedHandlerProvider` is already managing that